src/Umbraco.Web.Common/Extensions/UrlHelperExtensions.cs

```diff
@@ -21,6 +21,236 @@ namespace Umbraco.Extensions;
+                nameof(actionName));
+        }
+
+        if (controllerName == null)
+        {
+            throw new ArgumentNullException(nameof(controllerName));
+        }
+
+        if (string.IsNullOrWhiteSpace(controllerName))
+        {
+            throw new ArgumentException(
+                "Value can't be empty or consist only of white-space characters.",
+                nameof(controllerName));
+        }
+
+        if (area.IsNullOrWhiteSpace())
+        {
+            if (id == null)
+            {
+                return url.Action(actionName, controllerName);
+            }
+
+            return url.Action(actionName, controllerName, new { id });
+        }
+
+        if (id == null)
+        {
+            return url.Action(actionName, controllerName, new { area });
+        }
+
+        return url.Action(actionName, controllerName, new { area, id });
+    }
+
+    /// <summary>
+    ///     Return the Base Url (not including the action) for a Web Api service
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <param name="url"></param>
+    /// <param name="umbracoApiControllerTypeCollection"></param>
+    /// <param name="actionName"></param>
+    /// <returns></returns>
+    public static string? GetUmbracoApiServiceBaseUrl<T>(
+        this IUrlHelper url,
+        UmbracoApiControllerTypeCollection umbracoApiControllerTypeCollection,
+        string actionName)
+        where T : UmbracoApiController =>
+        url.GetUmbracoApiService<T>(umbracoApiControllerTypeCollection, actionName)?.TrimEnd(actionName);
+
+    public static string? GetUmbracoApiServiceBaseUrl<T>(
+        this IUrlHelper url,
+        UmbracoApiControllerTypeCollection umbracoApiControllerTypeCollection,
+        Expression<Func<T, object>> methodSelector)
+        where T : UmbracoApiController
+    {
+        MethodInfo? method = ExpressionHelper.GetMethodInfo(methodSelector);
+        if (method == null)
+        {
+            throw new MissingMethodException("Could not find the method " + methodSelector + " on type " + typeof(T) +
+                                             " or the result ");
+        }
+
+        return url.GetUmbracoApiService<T>(umbracoApiControllerTypeCollection, method.Name)?.TrimEnd(method.Name);
+    }
+
+    /// <summary>
+    ///     Return the Url for an action with a cache-busting hash appended
+    /// </summary>
+    /// <returns></returns>
+    public static string GetUrlWithCacheBust(
+        this IUrlHelper url,
+        string actionName,
+        string controllerName,
+        RouteValueDictionary routeVals,
+        IHostingEnvironment hostingEnvironment,
+        IUmbracoVersion umbracoVersion)
+    {
+        var applicationJs = url.Action(actionName, controllerName, routeVals);
+        applicationJs = applicationJs + "?umb__rnd=" +
+                        GetCacheBustHash(hostingEnvironment, umbracoVersion);
+        return applicationJs;
+    }
```

Comment:
Not sure if this should be reintroduced either - it's unused by the core. On the other hand, it seems to be harmless and if it can alleviate some breakage out there, I guess there's no harm in reintroducing it.